cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/etl/transform/ScriptFilterTransform.java

```diff
@@ -98,6 +102,10 @@ private void init() {
     ScriptEngineManager manager = new ScriptEngineManager();
     engine = manager.getEngineByName("JavaScript");
 
+    // initialize metrics and logging
+    engine.put("metrics", metrics);
+    engine.put("LOG", logger);
```

Comment:
In java, because `LOG` is a global static while `metrics` is a class field. Since this is Javascript, we can have a convention saying all global variables available to the script is in all caps. Better yet, we can namespace the CDAP variables by a `CDAP` namespace:

```java
  engine.put("CDAP", new ScriptGlobal());
  engine.eval("CDAP.logger.info('Hello, World')");

  public static final class ScriptGlobal {
    public final Logger logger;
    public final Metrics metrics;
   ...
  }
```